sale_delivery_date/models/sale_order_line.py

```diff
@@ -6,124 +6,201 @@
-            # the original planned date (but change the time), the transfer
-            # will be considered as "late"
-            keep_same_day=bool(self.order_id.commitment_date),
+    def _prepare_procurement_values_commitment_date(self, res):
+        """Set 'date_planned' according to 'commitment_date'.
+
+        We want to find the date when we could start working on the transfers
+        so we will be able to ship the goods to the customer in respect to the
+        `commitment_date`.
+        The 'date_deadline' is already set to the 'commitment_date' (that should
+        preferably fit the partner's time window if any).
+        """
+        warehouse = self.order_id.warehouse_id
+        partner = self.order_id.partner_id
+        __, security_lead, workload = self._get_delays()
+        worload_days = self._delay_to_days(workload)
+        # Find the first date in the WH calendar by going back in the past, if any.
+        res["date_planned"] = self._deduce_workload_and_security_lead(
+            res["date_deadline"], partner, warehouse, worload_days, security_lead
         )
-        if new_date_planned:
-            res["date_planned"] = new_date_planned
         return res
 
-    def _warehouse_calendar_prepare_procurement_values(self, res):
-        date_planned = res.get("date_planned")
-        calendar = self.order_id.warehouse_id.calendar_id
-        if date_planned and calendar:
-            customer_lead, security_lead, workload = self._get_delays()
-            # plan_days() expect a number of days instead of a delay
-            workload_days = self._delay_to_days(workload)
-            td_workload = timedelta(days=workload)
-            # Remove the workload that has been added by odoo
-            date_planned -= td_workload
-            # Add the workload, with respect to the wh calendar
-            res["date_planned"] = calendar.plan_days(
-                workload_days, date_planned, compute_leaves=True
-            )
-        return res
+    def _prepare_procurement_values_no_commitment_date(self, res, date_from=None):
+        """Set 'date_planned' and 'date_deadline' if no 'commitment_date'.
 
-    def _delivery_window_prepare_procurement_values(self, res):
-        date_planned = res.get("date_planned")
-        if not date_planned:
-            return res
-        new_date_planned = self._prepare_procurement_values_time_windows(
-            fields.Datetime.to_datetime(date_planned)
+        If date_from is set, use it as the starting date to compute delivery dates,
+        otherwise, use order's date_order.
+        """
+        customer_lead, security_lead, workload = self._get_delays()
+        workload_days = self._delay_to_days(workload)
+        if not date_from:
+            date_from = self.order_id.date_order
+        date_planned = (
+            date_from
+            + timedelta(days=customer_lead or 0.0)
+            - timedelta(days=security_lead)
         )
```

Comment:
According to me, the lead should be removed
```suggestion
        date_planned = date_from
```
workload = customer_lead - security_lead
So you apply twice the workload, once here and once below taking care of the calendar.